rustls-cert-gen/src/cert/params.rs

```diff
@@ -0,0 +1,146 @@
+use rcgen::{CertificateParams, DistinguishedName};
+
+use super::ca::CaParams;
+use super::entity::EndEntityParams;
+use super::signature::Signature;
+
+/// Builder to configure TLS [CertificateParams] to be finalized
+/// into either a Ca or an End-Entity.
+pub struct BuildParams {
+	params: CertificateParams,
+}
+
+impl Default for BuildParams {
+	fn default() -> Self {
+		Self::new()
+	}
+}
+
+impl BuildParams {
+	/// Initialize `CertificateParams` with defaults
+	/// # Example
+	/// ```
+	/// # use rustls_cert_gen::BuildParams;
+	/// let cert = BuildParams::new();
+	/// ```
+	pub fn new() -> Self {
+		let mut params = CertificateParams::default();
+		// override default Common Name
+		let distinguished_name = DistinguishedName::new();
+		params.distinguished_name = distinguished_name;
+		Self { params }
+	}
+	/// Set signature algorithm (instead of default). Returns `crate::Result<Self>`.
+	/// # Example
+	/// ```
+	/// # use rustls_cert_gen::BuildParams;
+	/// let cert = BuildParams::new().sig_algo("pkcs_ed25519");
+	/// ```
+	pub fn sig_algo(mut self, alg: &str) -> crate::Result<Self> {
```

Comment:
nit: let's expand this name:
```suggestion
	pub fn signature_algorithm(mut self, alg: &str) -> crate::Result<Self> {
```